tripal_chado/src/Plugin/Field/FieldType/ChadoAdditionalTypeTypeDefault.php

```diff
@@ -394,4 +465,99 @@ public function isCompatible(TripalEntityType $entity_type) : bool {
     return $compatible;
   }
 
+  /**
+   * {@inheritDoc}
+   * @see \Drupal\tripal\TripalField\Interfaces\TripalFieldItemInterface::discover()
+   */
+  public static function discover(TripalEntityType $bundle, string $field_id, array $field_definitions) : array {
+
+    /** @var \Drupal\tripal_chado\Database\ChadoConnection $chado **/
+    $chado = \Drupal::service('tripal_chado.database');
+    $schema = $chado->schema();
+
+    // Initialize with an empty field list.
+    $field_list = [];
+
+    // Make sure the base table setting exists.
+    $base_table = $bundle->getThirdPartySetting('tripal', 'chado_base_table');
+    if (!$base_table) {
+      return $field_list;
+    }
+
+    // For this field, we need either a "type_id" column in the base table,
+    // or else have it specified in a property table. Sometimes we have both.
+    $type_table = NULL;
+    $type_column = NULL;
+    $base_table_def = $schema->getTableDef($base_table, ['format' => 'Drupal']);
+    $base_type_column = 'type_id';
+    $base_type_id = $base_table_def['fields'][$base_type_column] ?? NULL;
+    $prop_type_id = NULL;
+    if ($base_type_id) {
+      $type_table = $base_table;
+      $type_column = $base_type_column;
+    }
+    else {
+      $prop_table = $base_table . 'prop';
+      $prop_type_column = 'type_id';
+      if ($chado->schema()->tableExists($prop_table)) {
+        $prop_table_def = $schema->getTableDef($prop_table, ['format' => 'Drupal']);
+        $prop_type_id = $prop_table_def['fields'][$prop_type_column] ?? NULL;
+        if ($prop_type_id) {
+          $type_table = $prop_table;
+          $type_column = $prop_type_column;
+        }
+      }
+    }
+
+    // If neither of these two type_ids are present, then this field
+    // is not discoverable for the base table.
+    if (!$type_table) {
+      return $field_list;
+    }
+
+    // Create a field entry in the list
+    $termIdSpace = $bundle->getTermIdSpace();
+    $termAccession = $bundle->getTermAccession();
+    $fixed_value = $termIdSpace . ':' . $termAccession;
+    $field_list[] = [
+      'name' => self::generateFieldName($bundle, 'type', 0),
+      'content_type' => $bundle->getID(),
+      'label' => 'Type',
+      'type' => self::$id,
+      'description' => 'This field specifies the controlled vocabulary term'
+          . ' for this content type as "' . $fixed_value . '"',
+      'cardinality' => 1,
+      'required' => TRUE,
+      'storage_settings' => [
+        'storage_plugin_id' => 'chado_storage',
+        'storage_plugin_settings' => [
+          'base_table' => $base_table,
+          'type_table' => $type_table,
+          'type_column' => $type_column,
+        ],
+      ],
+      'settings' => [
+        'termIdSpace' => $termIdSpace,
+        'termAccession' => $termAccession,
```

Comment:
The newly discovered field should have the fixed value set by default. This allows that checkbox to be checked by default and makes the type table/column be automatically detected when the type field is added by check fields rather then requiring an edit to the field.

I know there are cases where this field doesn't define the content type but that feels like the exception rather then the default 🤔 and you do specifically mention the fixed value in the field description you generate :wink'

```suggestion
        'termAccession' => $termAccession,
        'fixed_value' => $fixed_value,
```

Reply:
I removed the default of checked with commit https://github.com/tripal/tripal/pull/1992/commits/0bdb0f8801a19702fb33a4d6ec9b857e085e2147
My thinking was that this was a safer default, however on further consideration I agree with your thinking that it is a sensible default, as those other use cases will be uncommon.